From 931f7f918e5cde0655331f26d45d599c231b6926 Mon Sep 17 00:00:00 2001
From: Vaibhav Devmurari <vdevmurari@google.com>
Date: Mon, 9 Dec 2024 13:15:03 +0000
Subject: [PATCH] Fix: Block opening settings app on keyguard without user auth

Test: atest KeyGestureEventTests
Bug: 378900798
Flag: EXEMPT bugfix
(cherry picked from commit d615298466085c4a88c6733804160e0c1ee7e31e)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:e4d483a2ef99a71c6fcd6ad2e6c2f8f88ba380f4)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:e832d2b23784ede62259fb7518573e12dced2209)
Merged-In: I89d43872108710d20e0c4ef7e652d389896155d7
Change-Id: I89d43872108710d20e0c4ef7e652d389896155d7
---
 .../core/java/com/android/server/policy/PhoneWindowManager.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index 339428260800..60b0d7856ac9 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -3110,7 +3110,7 @@ public class PhoneWindowManager implements WindowManagerPolicy {
                 }
                 break;
             case KeyEvent.KEYCODE_I:
-                if (firstDown && event.isMetaPressed()) {
+                if (firstDown && event.isMetaPressed() && isUserSetupComplete() && !keyguardOn) {
                     showSystemSettings();
                     logKeyboardSystemsEvent(event, KeyboardLogEvent.LAUNCH_SYSTEM_SETTINGS);
                     return true;
-- 
2.34.1

